src/lib.rs

```diff
@@ -90,6 +100,60 @@ pub fn generate_next_shielding_payment_address(
     }
 }
 
+fn handle_block(
+    tree: &mut CommitmentTree<Node>,
+    block_data: String,
```

Comment:
```suggestion
    block_data: &str,
```
Change to &str so we don't have to use the owned data type